compiler/index.js

```diff
@@ -1,7 +1,62 @@
-var handlebars = require('handlebars')
+var Handlebars = require('handlebars');
+var walkSync = require('walk-sync');
+var fs = require('fs');
+var path = require('path');
 
 function Compiler(rootPath) {
-  this.root = rootPath;
+  this.rootPath = rootPath;
 }
 
+
+Compiler.prototype.collectInputFilePaths = function(dirPath, type) {
+  var inputPath = path.join(this.rootPath, dirPath);
+  var filePaths = walkSync(inputPath).filter(function(filePath) {
+    return filePath.substr(filePath.length - type.length) === type;
+  });
+
+  return filePaths;
+};
+
+Compiler.prototype.registerHelper = function(filePath) {
+  var helperPath = path.join(this.rootPath + filePath);
+  var helperName = path.basename(helperPath, '.js');
```

Comment:
this could potentially cause issue with nested helpers because `helpers/nested/t-helper.js' and 'helpers/t-helper.js' would be registered with the same name (`t-helper`). Souldn't trim the path to the file, just the extension.